types/tx.go

```diff
@@ -6,6 +6,13 @@ import (
 
 type Tx []byte
 
+// NOTE: this is the hash of the go-wire encoded Tx.
+// Tx has no types at this level, so just length-prefixed.
+// Maybe it should just be the hash of the bytes tho?
```

Comment:
Alternatively, it might make sense to add types here, and let `[]byte` be `type(0x1)` so we get some future proofing? Versioning data structures is actually a great use case of go-wire!